src/penalty_parameter_functions.jl

```diff
@@ -164,6 +165,10 @@ function is_variable_dependent(::Type{ScalarPenaltyParameter})::Bool
     return false
 end
 
+function reset_penalty_parameter(r::ScalarPenaltyParameter)::ScalarPenaltyParameter
```

Comment:
I think the best way to go with this is actually to implement
```
function reset_penalty_parameter(r::AbstractPenaltyParameter)
    return # intentional no-op
end
```
and let that handle any case that doesn't need a reset function.